### PR TITLE
Merge the Customer endpoints into one domain PR

### PR DESCRIPTION
**Pull request by PrestaEdit**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Branch?           | dev
| Description?      | Consolidates the pending Customer PRs into one domain PR, and drops one endpoint the team had already flagged as a duplicate
| Type?             | new feature
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Related to PrestaShop/PrestaShop#39630
| How to test?      | See below
| Sponsor company   | PrestaEdit

## What this PR does

Merges #218, #225, #243 and #342 into one PR, following the mutualisation done on the Product domain in #410.

### Endpoints added

| Method | URI | CQRS | Scope |
|--------|-----|------|-------|
| PATCH | `/customers/{customerId}/private-notes` | `SetPrivateNoteAboutCustomerCommand` | `customer_write` |
| GET | `/customers/{customerId}/orders` | `GetCustomerOrders` | `customer_read` |
| GET | `/customers/{customerId}/carts` | `GetCustomerCarts` | `customer_read` |
| PUT | `/customers/{customerId}/transform-to-customers` | `TransformGuestToCustomerCommand` | `customer_write` |

#218 and #225 both wrote `tests/Integration/ApiPlatform/CustomerEndpointTest.php` and conflict on cherry-pick — the usual sign that they belonged in one PR.

### #342 is dropped, not merged

It exposed `GetCustomerForAddressCreation` as `GET /customers/address-creation-infos?customerEmail=`, returning `{customerId, firstName, lastName, company}`.

That query is **already listed in `EXCLUDED_CQRS_CLASSES` on `dev`**, with the `USELESS_DUPLICATE` reason — the team had recorded the decision not to expose it. The exclusion holds up: `GET /customers/search` takes `phrases[]` that match *"first name, last name, email, company name and id"* and returns `idCustomer`, `firstname`, `lastname`, `company` and more — a strict superset of the four fields, reachable by the same email lookup.

Since the exclusion entry already exists on `dev`, this PR does not need to add anything for it; the resource and its test are simply not carried over.

### Tests

**The private-note test now asserts its work.** It used to check the `204` and stop there, because the write had no read side inside its own PR:

```php
$this->assertNull($return);
```

`GET /customers/{customerId}/details` — already on `dev` — returns the note as `generalInformation.privateNote`, so the test now reads it back.

`CustomerTransformGuestEndpointTest` is kept as its own class: it already builds every fixture through `POST /customers` and asserts through `GET /customers/{id}`, and it needs a `customer` / `customer_group` reset that the main suite does not want.

## How to test

```
PATCH /customers/{id}/private-notes         -> 204, then the details show the note
GET   /customers/{id}/orders                -> 200 {customerId, orders}
GET   /customers/{id}/carts                 -> 200 {customerId, carts}
PUT   /customers/{id}/transform-to-customers -> 204, then the GET shows guest: false
```

Covered by `CustomerEndpointTest` and `CustomerTransformGuestEndpointTest`.

## Supersedes

- #218
- #225
- #243

and closes #342 as a duplicate (its query is already excluded on `dev`).

All four will be closed once the CI is green here.

### CODE REVIEW

**Review by mattgoud — CHANGES_REQUESTED** [review]
Reviewed against a local install with the branch mounted and every endpoint called. The error mapping is the best of the three Shop/ImageType/Customer merges I have looked at today: `422` on the transformation of a non-guest, `404` on an unknown customer id for all four endpoints, and the `Assert\NotNull` does fire on the private note. One blocking point, and a few things worth tightening.

## Blocking: two routes are reserved and answer nothing

```
_api_/customer_private_notes/{customerId}{._format}_get        GET  /customer_private_notes/{customerId}.{_format}
_api_/transform_guest_to_customers/{customerId}{._format}_get  GET  /transform_guest_to_customers/{customerId}.{_format}

_controller: api_platform.action.not_exposed()
```

```
GET /admin-api/customer_private_notes/2
404 {"detail":"This route does not aim to be called.","class":"ApiPlatform\\Metadata\\Exception\\NotExposedHttpException"}
```

`CustomerPrivateNote` and `TransformGuestToCustomer` both carry `#[ApiProperty(identifier: true)]` while exposing only a `PATCH` and a `PUT`. With no item `GET` declared, API Platform registers one itself so the resource stays addressable, and it answers that 404. Two public URIs end up reserved for nothing, and they are snake_case on top of it, which the URI convention of this repository does not allow anywhere else.

Removing the attribute from both classes is enough here, and I checked that it costs nothing:

```
routes after removal   only /customers/{customerId}/private-notes and /customers/{customerId}/transform-to-customers remain

PATCH /customers/2/private-notes            204   and ps_customer.note really holds the new value
PUT   /customers/2/transform-to-customers   422   still, on a customer who is not a guest
PATCH /customers/999999/private-notes       404   still
GET   /customer_private_notes/2             404   and now it is a real "no route", not a reserved one
```

Worth noting for next time: this is not the same fix as the one on PrestaShop/ps_apiresources#422. There the property is named `id`, which API Platform treats as the identifier by convention, so it needed an explicit `identifier: false`. Here the property is `$customerId`, so dropping the attribute is enough.

## Worth fixing

**Two ways to say "no private note", two different answers.**

```
PATCH /customers/2/private-notes   {}                     422   privateNote: This value should not be null.
PATCH /customers/2/private-notes   {"privateNote":null}   400   The type of the "privateNote" attribute ... must be string, NULL given
```

Omitting the field gives the clean validation error, sending an explicit `null` hits the denormalizer first and surfaces its message. Typing the property `?string` and keeping the `NotNull` would route both through the same 422.

**Two semantics an API consumer will misread, and neither is visible from the payload.**

`totalPaid` is `total_paid_real`, the amount actually received, not the order total. On the demo fixtures it reads `"€0.00"` on every order while `ps_orders.total_paid` holds `61.80`, `169.90` and so on. That is what `GetCustomerOrdersHandler` passes and it matches the BO, so nothing to change in the behaviour, but a caller reading `totalPaid` will assume it is the order total.

`/customers/{id}/carts` excludes carts that already became an order, because the core calls `Cart::getCustomerCarts($id, false)`. On a customer with five carts, all five converted, the endpoint answers `[]`. That is also the real reason the test can only ever cover the empty case, which is worth writing down next to it.

A one-line docblock on each of the two properties is enough.

**`requirements: ['customerId' => '\d+']`** is on `CustomerPrivateNote`, `TransformGuestToCustomer`, `Customer` and `CustomerDetails`, but not on `CustomerCart` and `CustomerOrder`.

## Tests

There is no `testInvalid*` in either class. One asserting `422` on `PATCH {}` would lock in the behaviour I verified above, and a second on the `null` payload would lock in whichever answer you settle on for it.

## On the automated pre-review

Its main ask was to verify the field alignment against the core, which it could not do. Done, and it is exact on all nine properties, names and types:

- `CartSummary::getCartId(): int`, `getCreationDate(): string`, `getTotalPrice(): string`
- `OrderSummary::getOrderId(): int`, `getOrderPlacedDate(): string`, `getPaymentMethodName(): string`, `getOrderStatus(): string`, `getOrderProductsCount(): int`, `getTotalPaid(): string`

`customerId` is not an orphan either, the live response carries it on every row.

Three of its other points I would set aside:

- **`DateTimeImmutable` for the two date properties**: no. The core returns them already formatted as strings, so `string` is the correct type. The format is `2026-09-18 09:44:06` rather than ISO-8601, but that comes from the core and changing it here would not be faithful.
- **`transform-to-customers` in `ApiResourceUriTemplateRector::SKIPPED_KEYWORDS`**: not needed, the route registers exactly as declared and the Rector job is green.
- **The missing `validationContext`**: real, but it changes nothing today, and the report says so itself. `Assert\NotNull` carries no group, so it fires on the `PATCH` regardless, which the 422 above confirms.

**Comment by mattgoud** [comment]
**Pre-QA OK**

I pushed the review fixes on this branch myself, so this is a pre-QA: same process as a QA, but I am not setting the QA label. Another QA has to sign off.

Tested on: core `9.2.x` at `8103e04c6b7`, module at `9d23e65`, Docker dev env, shop installed from 9.2.0 (`PS_VERSION_DB` 9.2.0)
Shop: single shop, demo data
Client: `curl` against `http://localhost:8001/admin-api`, API client with `customer_read customer_write`

No linked issue: the How to test section of the description was concrete enough to test directly.

### Compared states

Both states come from this branch. **Without the fix** = the four resource classes put back to their state before the review commit (`git checkout HEAD~1 -- src/ApiPlatform/Resources/Customer/`), Symfony cache deleted between switches. The states really differ: `debug:router --app-id=admin-api` lists 6 Customer routes without the fix and 4 with it.

### Result

| Call | Without the fix | With the fix |
|---|---|---|
| `GET /customer_private_notes/2` | 404 `NotExposedHttpException` (reserved route) | 404 `No route found` |
| `GET /transform_guest_to_customers/2` | 404 `NotExposedHttpException` (reserved route) | 404 `No route found` |
| `PATCH /customers/2/private-notes` `{"privateNote":"QA 428 note"}` | 204 | 204, `ps_customer.note` = `'QA 428 note'`, `/details` returns it |
| `PATCH .../private-notes` `{}` | 422 `privateNote: This value should not be null.` | 422, same violation |
| `PATCH .../private-notes` `{"privateNote":null}` | **400** `must be one of "string" ("null" given)` | **422**, same violation as `{}` |
| `PATCH .../private-notes` `{"privateNote":""}` | 204 (clears the note) | 204 (clears the note) |
| `PATCH /customers/999999/private-notes` | 404 | 404 |
| `GET /customers/2/orders` | 200, 5 orders | 200, 5 orders |
| `GET /customers/abc/orders` | **500** `Customer id 0 is invalid.` | **404** `No route found` |
| `GET /customers/2/carts` | 200 `[]` | 200 `[]` |
| `GET /customers/abc/carts` | **500** `Customer id 0 is invalid.` | **404** `No route found` |
| `GET /customers/999999/orders` and `/carts` | 404 | 404 |
| `PUT /customers/2/transform-to-customers` (registered customer) | 422 `already exists as non-guest` | 422, same |
| `PUT /customers/999999/transform-to-customers` | 404 | 404 |
| `POST /customers` guest, then `PUT .../transform-to-customers` | 204, `guest` true → false | 204, `guest` true → false |
| Same `PUT` a second time | 422 | 422 |
| `GET /customers/2/orders` without token | 401 | 401 |

The missing `customerId` requirement on the two collections was worse than an inconsistency: without it a non numeric id reached the handler and answered 500. It now stops at routing.

The two semantics documented in the review commit hold on the demo data:

- `totalPaid` is `€0.00` on all 5 orders of customer 2, while `ps_orders.total_paid` holds `61.80`, `169.90`, `14.90`, `14.90` and `20.90`. It is `total_paid_real`, as the new docblock says.
- `/customers/2/carts` answers `[]` although customer 2 has 5 carts: each of them became order 1 to 5.

### Automated tests

- CI ✅ on `9d23e65`: 19 checks green, including the 8 integration jobs (PHP 8.1 to 8.5 against core 9.0.3, 9.1.5, 9.2.x and develop).
- `CustomerEndpointTest` and `CustomerTransformGuestEndpointTest` run locally in the container ✅: 55 tests, 307 assertions. `testInvalidCustomerPrivateNote` covers both `{}` and `{"privateNote":null}`.

### Before merge

- The review from the 21st is still `CHANGES_REQUESTED`. As co-author of the fix it answers, I will not be the one approving it: this needs a second reviewer.
- #218, #225, #243 and #342 are still open, to close once this is merged, as the description says.

<details>
<summary>Raw transcripts</summary>

Without the fix:

```
phantom item route (private note)            GET    /customer_private_notes/2                      404  {"detail": "This route does not aim to be called.", "class": "ApiPlatform\\Metadata\\Exception\\NotExposedHttpException"}
phantom item route (transform)               GET    /transform_guest_to_customers/2                404  {"detail": "This route does not aim to be called.", "class": "ApiPlatform\\Metadata\\Exception\\NotExposedHttpException"}
set private note                             PATCH  /customers/2/private-notes                     204  
  -> details.generalInformation.privateNote = "QA 428 note"
private note, field omitted                  PATCH  /customers/2/private-notes                     422  [{"propertyPath": "privateNote", "message": "This value should not be null.", "code": "ad32d13f-c3d4-423b-909a-857b961eb720"}]
private note, explicit null                  PATCH  /customers/2/private-notes                     400  {"detail": "The type of the \"privateNote\" attribute for class \"PrestaShop\\Module\\APIResources\\ApiPlatform\\Resources\\Customer\\CustomerPrivateNote\" must be one of \"string\" (\"null\" given).", "class": "Symfony\\Component\\Serializer\\Exception\\NotNormalizableValueException"}
private note, empty string                   PATCH  /customers/2/private-notes                     204  
private note, unknown customer               PATCH  /customers/999999/private-notes                404  {"detail": "Customer with id \"999999\" was not found.", "class": "PrestaShop\\PrestaShop\\Core\\Domain\\Customer\\Exception\\CustomerNotFoundException"}
orders                                       GET    /customers/2/orders                            200  [{"customerId": 2, "orderId": 1, "orderPlacedDate": "2026-09-23 15:32:45", "paymentMethodName": "Payment by check", "orderStatus": "Canceled", "orderProductsCount": 2, "totalPaid": "€0.00"}, {"customerId": 2, "orderId": 2, "orderPlacedDate": "2026-09-23 15:32:45", "paymentMethodName": "Payment by ch
orders, unknown customer                     GET    /customers/999999/orders                       404  {"detail": "Customer with id \"999999\" was not found.", "class": "PrestaShop\\PrestaShop\\Core\\Domain\\Customer\\Exception\\CustomerNotFoundException"}
orders, non numeric id                       GET    /customers/abc/orders                          500  {"detail": "Customer id 0 is invalid.", "class": "PrestaShop\\PrestaShop\\Core\\Domain\\Customer\\Exception\\CustomerConstraintException"}
carts (5 carts, all ordered)                 GET    /customers/2/carts                             200  []
carts, unknown customer                      GET    /customers/999999/carts                        404  {"detail": "Customer with id \"999999\" was not found.", "class": "PrestaShop\\PrestaShop\\Core\\Domain\\Customer\\Exception\\CustomerNotFoundException"}
carts, non numeric id                        GET    /customers/abc/carts                           500  {"detail": "Customer id 0 is invalid.", "class": "PrestaShop\\PrestaShop\\Core\\Domain\\Customer\\Exception\\CustomerConstraintException"}
transform a registered customer              PUT    /customers/2/transform-to-customers            422  {"detail": "Customer with id \"2\" already exists as non-guest", "class": "PrestaShop\\PrestaShop\\Core\\Domain\\Customer\\Exception\\CustomerTransformationException"}
transform, unknown customer                  PUT    /customers/999999/transform-to-customers       404  {"detail": "Customer with id \"999999\" was not found", "class": "PrestaShop\\PrestaShop\\Core\\Domain\\Customer\\Exception\\CustomerNotFoundException"}
  -> guest created: id 4, guest=True
transform the guest                          PUT    /customers/4/transform-to-customers            204  
  -> after transform: guest=False
transform it again                           PUT    /customers/4/transform-to-customers            422  {"detail": "Customer with id \"4\" already exists as non-guest", "class": "PrestaShop\\PrestaShop\\Core\\Domain\\Customer\\Exception\\CustomerTransformationException"}
orders, no token                             GET    /customers/2/orders                            401  "No Authorization header provided"
```

With the fix:

```
phantom item route (private note)            GET    /customer_private_notes/2                      404  {"detail": "No route found for \"GET http://localhost:8001/admin-api/customer_private_notes/2\"", "class": "Symfony\\Component\\HttpKernel\\Exception\\NotFoundHttpException"}
phantom item route (transform)               GET    /transform_guest_to_customers/2                404  {"detail": "No route found for \"GET http://localhost:8001/admin-api/transform_guest_to_customers/2\"", "class": "Symfony\\Component\\HttpKernel\\Exception\\NotFoundHttpException"}
set private note                             PATCH  /customers/2/private-notes                     204  
  -> details.generalInformation.privateNote = "QA 428 note"
private note, field omitted                  PATCH  /customers/2/private-notes                     422  [{"propertyPath": "privateNote", "message": "This value should not be null.", "code": "ad32d13f-c3d4-423b-909a-857b961eb720"}]
private note, explicit null                  PATCH  /customers/2/private-notes                     422  [{"propertyPath": "privateNote", "message": "This value should not be null.", "code": "ad32d13f-c3d4-423b-909a-857b961eb720"}]
private note, empty string                   PATCH  /customers/2/private-notes                     204  
private note, unknown customer               PATCH  /customers/999999/private-notes                404  {"detail": "Customer with id \"999999\" was not found.", "class": "PrestaShop\\PrestaShop\\Core\\Domain\\Customer\\Exception\\CustomerNotFoundException"}
orders                                       GET    /customers/2/orders                            200  [{"customerId": 2, "orderId": 1, "orderPlacedDate": "2026-09-23 15:32:45", "paymentMethodName": "Payment by check", "orderStatus": "Canceled", "orderProductsCount": 2, "totalPaid": "€0.00"}, {"customerId": 2, "orderId": 2, "orderPlacedDate": "2026-09-23 15:32:45", "paymentMethodName": "Payment by ch
orders, unknown customer                     GET    /customers/999999/orders                       404  {"detail": "Customer with id \"999999\" was not found.", "class": "PrestaShop\\PrestaShop\\Core\\Domain\\Customer\\Exception\\CustomerNotFoundException"}
orders, non numeric id                       GET    /customers/abc/orders                          404  {"detail": "No route found for \"GET http://localhost:8001/admin-api/customers/abc/orders\"", "class": "Symfony\\Component\\HttpKernel\\Exception\\NotFoundHttpException"}
carts (5 carts, all ordered)                 GET    /customers/2/carts                             200  []
carts, unknown customer                      GET    /customers/999999/carts                        404  {"detail": "Customer with id \"999999\" was not found.", "class": "PrestaShop\\PrestaShop\\Core\\Domain\\Customer\\Exception\\CustomerNotFoundException"}
carts, non numeric id                        GET    /customers/abc/carts                           404  {"detail": "No route found for \"GET http://localhost:8001/admin-api/customers/abc/carts\"", "class": "Symfony\\Component\\HttpKernel\\Exception\\NotFoundHttpException"}
transform a registered customer              PUT    /customers/2/transform-to-customers            422  {"detail": "Customer with id \"2\" already exists as non-guest", "class": "PrestaShop\\PrestaShop\\Core\\Domain\\Customer\\Exception\\CustomerTransformationException"}
transform, unknown customer                  PUT    /customers/999999/transform-to-customers       404  {"detail": "Customer with id \"999999\" was not found", "class": "PrestaShop\\PrestaShop\\Core\\Domain\\Customer\\Exception\\CustomerNotFoundException"}
  -> guest created: id 3, guest=True
transform the guest                          PUT    /customers/3/transform-to-customers            204  
  -> after transform: guest=False
transform it again                           PUT    /customers/3/transform-to-customers            422  {"detail": "Customer with id \"3\" already exists as non-guest", "class": "PrestaShop\\PrestaShop\\Core\\Domain\\Customer\\Exception\\CustomerTransformationException"}
orders, no token                             GET    /customers/2/orders                            401  "No Authorization header provided"
```

</details>

**Comment by mattgoud** [comment]
**Pre-QA, Swagger evidence** (follow-up to the pre-QA comment above)

Same two states as above, replayed through the back-office Swagger UI (`Configure > Advanced > Admin API` docs), authorized with a `customer_read customer_write` client. Only the two calls whose answer changes between the states are shown; every other call answers the same in both, as listed in the table above.

**`PATCH /customers/2/private-notes` with `{"privateNote": null}`**: 400 from the denormalizer without the fix, the same 422 violation as an omitted field with it.

note-null

**`GET /customers/abc/orders`**: 500 `Customer id 0 is invalid.` without the fix, 404 at routing with it. `/customers/abc/carts` behaves the same.

orders-abc

The two reserved routes (`/customer_private_notes/{id}`, `/transform_guest_to_customers/{id}`) cannot be shown in Swagger: API Platform keeps not exposed operations out of the OpenAPI document in both states. The route listing in the comment above is the evidence for that part.

The Swagger curl block was hidden before capturing, and the BO `_token` echoed in the 404 detail is masked.

**Comment by mattgoud** [comment]
**Pre-QA, Swagger evidence** (follow-up to the pre-QA comment above)

Replayed through the back-office Swagger UI (`Configure > Advanced > Admin API` docs) on the same 9.2.0 shop, authorized with a `customer_read customer_write` client.

### The 4 endpoints, with the fix

`PATCH /customers/2/private-notes` answers 204. `GET /customers/2/orders` answers 200 with the 5 orders of customer 2 (`totalPaid` is `total_paid_real`, hence `€0.00`).

nominal-1

`GET /customers/2/carts` answers 200 `[]`: customer 2 has 5 carts, and each became an order. `PUT /customers/5/transform-to-customers` on a guest created through `POST /customers` answers 204.

nominal-2

Read back with `GET /customers/5`: `guest` is now `false`.

nominal-3

### What changes between the two states

Without the fix = the four resource classes put back to their state before the review commit, cache deleted, same page and authorization.

**`PATCH /customers/2/private-notes` with `{"privateNote": null}`**: 400 from the denormalizer without the fix, the same 422 violation as an omitted field with it.

note-null

**`GET /customers/abc/orders`** and **`GET /customers/abc/carts`**: 500 `Customer id 0 is invalid.` without the fix, 404 at routing with it.

orders-abc

carts-abc

Every other call answers the same in both states, as listed in the table of the comment above. The two reserved routes (`/customer_private_notes/{id}`, `/transform_guest_to_customers/{id}`) cannot be shown in Swagger, since API Platform keeps not exposed operations out of the OpenAPI document in both states: the route listing above is the evidence for that part.

The curl block was hidden before capturing, response headers are cropped out, and the BO `_token` echoed in one 404 detail is masked.

**Comment by mattgoud** [comment]
thanks @PrestaEdit for merging the Customer PRs into one, and for writing down why #342 was dropped.

I took the last round of my review myself to move this forward: rebased on `dev` (the branch was 26 commits behind) and pushed 9d23e65 on top of your commits. heads up, that was a force push, so reset your local branch before pushing anything else.

9d23e65 covers every point of my review:
- no more `identifier: true` on `CustomerPrivateNote` and `TransformGuestToCustomer`, the two reserved routes are gone
- `privateNote` is `?string`, an explicit null now gets the same 422 as an omitted field
- `customerId` requirement on the orders and carts collections, a non numeric id was a 500 and is now a 404
- docblocks on `totalPaid` and on the carts filter
- `testInvalidCustomerPrivateNote` covers both `{}` and `null`

evidence is in the two pre-QA comments above. since I wrote the fix, I'm dismissing my review instead of approving, so this needs another reviewer.

File: src/ApiPlatform/Resources/Customer/TransformGuestToCustomer.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+use PrestaShop\PrestaShop\Core\Domain\Customer\Command\TransformGuestToCustomerCommand;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Exception\CustomerNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Exception\CustomerTransformationException;
+use PrestaShopBundle\ApiPlatform\Metadata\CQRSUpdate;
+use Symfony\Component\HttpFoundation\Response;
+
+#[ApiResource(
+    operations: [
+        new CQRSUpdate(
+            uriTemplate: '/customers/{customerId}/transform-to-customers',
```
Comment (cnavarro-prestashop) [inline src/ApiPlatform/Resources/Customer/TransformGuestToCustomer.php]: Considering you transform one customer and not a batch of customers, it should be : 

```suggestion
            uriTemplate: '/customers/{customerId}/transform-to-customer',
```

I know that endpoints should be plural but you can add an exception into `SKIPPED_KEYWORDS` const : https://github.com/PrestaShop/ps_apiresources/blob/dcf05ff910c6473f0c98801a02becc891177d6ab/tests/Rector/ApiResourceUriTemplateRector.php#L102

File: src/ApiPlatform/Resources/Customer/CustomerPrivateNote.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+)]
+class CustomerPrivateNote
+{
+    public int $customerId;
+
+    /**
+     * Nullable so that an explicit null reaches the NotNull constraint and gets the same 422
+     * as an omitted field, instead of a 400 from the denormalizer.
+     */
+    #[Assert\NotNull]
```
Comment (cnavarro-prestashop) [inline src/ApiPlatform/Resources/Customer/CustomerPrivateNote.php]: In BO we apply `CleanHtml` constraint to a note. You should do the same thing here : 

```suggestion
    #[Assert\NotNull]
    #[CleanHtml]
```

File: src/ApiPlatform/Resources/Customer/CustomerCart.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    operations: [
+        new CQRSGetCollection(
+            uriTemplate: '/customers/{customerId}/carts',
+            requirements: ['customerId' => '\d+'],
+            CQRSQuery: GetCustomerCarts::class,
+            scopes: ['customer_read'],
+        ),
+    ],
+    exceptionToStatus: [
+        CustomerNotFoundException::class => Response::HTTP_NOT_FOUND,
```
Comment (cnavarro-prestashop) [inline src/ApiPlatform/Resources/Customer/CustomerCart.php]: Missing `CustomerConstraintException` mapping : 

```suggestion
        CustomerNotFoundException::class => Response::HTTP_NOT_FOUND,
        CustomerConstraintException::class => Response::HTTP_UNPROCESSABLE_ENTITY,
```

File: src/ApiPlatform/Resources/Customer/CustomerOrder.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    operations: [
+        new CQRSGetCollection(
+            uriTemplate: '/customers/{customerId}/orders',
+            requirements: ['customerId' => '\d+'],
+            CQRSQuery: GetCustomerOrders::class,
+            scopes: ['customer_read'],
+        ),
+    ],
+    exceptionToStatus: [
+        CustomerNotFoundException::class => Response::HTTP_NOT_FOUND,
```
Comment (cnavarro-prestashop) [inline src/ApiPlatform/Resources/Customer/CustomerOrder.php]: Missing `CustomerConstraintException` mapping : 

```suggestion
        CustomerNotFoundException::class => Response::HTTP_NOT_FOUND,
        CustomerConstraintException::class => Response::HTTP_UNPROCESSABLE_ENTITY,
```

File: tests/Integration/ApiPlatform/CustomerEndpointTest.php [hunk 5]
```diff
@@ -518,6 +533,85 @@ public function testGetCustomer(int $customerId): void
...
+
+    /**
+     * @depends testAddCustomer
+     */
+    public function testGetCustomerCarts(int $customerId): void
+    {
+        $response = $this->getItem('/customers/' . $customerId . '/carts', ['customer_read']);
+
+        // The query returns a list, so the endpoint is a collection: a freshly created customer
+        // has no carts yet, hence an empty list. Only the empty case can be covered here: the core
```
Comment (cnavarro-prestashop) [inline tests/Integration/ApiPlatform/CustomerEndpointTest.php]: It's false, in `setUpBeforeClass` you can create a cart assigned to a customer like we do for Order tests integration : https://github.com/PrestaShop/ps_apiresources/blob/dcf05ff910c6473f0c98801a02becc891177d6ab/tests/Integration/ApiPlatform/OrderReturnEndpointTest.php#L47

File: src/ApiPlatform/Resources/Customer/TransformGuestToCustomer.php [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+            requirements: ['customerId' => '\d+'],
+            output: false,
+            allowEmptyBody: true,
+            CQRSCommand: TransformGuestToCustomerCommand::class,
+            scopes: ['customer_write'],
```
Comment (cnavarro-prestashop) [inline src/ApiPlatform/Resources/Customer/TransformGuestToCustomer.php]: Suggested by AI pre-review : 
```suggestion
            scopes: ['customer_write'],
            validationContext: ['groups' => ['Default', 'Update']],
```

File: tests/Integration/ApiPlatform/CustomerTransformGuestEndpointTest.php [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+     * @depends testTransformGuestToCustomer
+     */
+    public function testTransformAlreadyRegisteredCustomerFails(int $customerId): void
+    {
+        // Transforming a customer that is no longer a guest is rejected
+        $this->updateItem(
+            '/customers/' . $customerId . '/transform-to-customers',
+            [],
+            ['customer_write'],
+            Response::HTTP_UNPROCESSABLE_ENTITY
```
Comment (cnavarro-prestashop) [inline tests/Integration/ApiPlatform/CustomerTransformGuestEndpointTest.php]: No test for 404 ?

File: tests/Integration/ApiPlatform/CustomerEndpointTest.php
Comment (cnavarro-prestashop) [inline tests/Integration/ApiPlatform/CustomerEndpointTest.php]: Not related to your modification but it should be `tearDownAfterClass` to reset tables after test.

**Review by cnavarro-prestashop — CHANGES_REQUESTED** [review]
